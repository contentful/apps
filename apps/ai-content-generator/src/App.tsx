import { useContext, useMemo } from 'react';
import { locations } from '@contentful/app-sdk';
import { useSDK } from '@contentful/react-apps-toolkit';
import ConfigScreen from '@locations/ConfigScreen';
import Dialog from '@locations/Dialog';
import Sidebar from '@locations/Sidebar';
import { SegmentAnalyticsContext } from '@providers/segmentAnalyticsProvider';

const ComponentLocationSettings = {
  [locations.LOCATION_APP_CONFIG]: ConfigScreen,
  [locations.LOCATION_DIALOG]: Dialog,
  [locations.LOCATION_ENTRY_SIDEBAR]: Sidebar,
};

const App = () => {
  const sdk = useSDK();
  const { identify } = useContext(SegmentAnalyticsContext);

  const Component = useMemo(() => {
    identify();

    for (const [location, component] of Object.entries(ComponentLocationSettings)) {
      if (sdk.location.is(location)) {
        return component;
      }
    }
  }, [sdk.location, identify]);

  return Component ? <Component /> : null;
};

export default App;
