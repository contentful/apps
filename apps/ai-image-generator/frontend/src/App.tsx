import { locations } from '@contentful/app-sdk';
import { useSDK } from '@contentful/react-apps-toolkit';
import { useMemo } from 'react';
import ConfigScreen from './locations/ConfigScreen';

const ComponentLocationSettings = {
  [locations.LOCATION_APP_CONFIG]: ConfigScreen,
};

const App = () => {
  const sdk = useSDK();

  const Component = useMemo(() => {
    for (const [location, component] of Object.entries(ComponentLocationSettings)) {
      if (sdk.location.is(location)) {
        return component;
      }
    }
  }, [sdk.location]);

  return Component ? <Component /> : null;
};

export default App;
