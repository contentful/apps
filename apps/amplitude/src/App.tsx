import { useMemo } from 'react';
import { locations } from '@contentful/app-sdk';
import { useSDK } from '@contentful/react-apps-toolkit';
import ConfigScreen from './locations/ConfigScreen';
import Sidebar from './locations/Sidebar';

const ComponentLocationSettings = {
  [locations.LOCATION_APP_CONFIG]: ConfigScreen,
  [locations.LOCATION_ENTRY_SIDEBAR]: Sidebar,
};

const App = () => {
  console.log('üöÄ Amplitude App: Initializing Amplitude integration app');
  
  const sdk = useSDK();
  
  console.log('üìç Amplitude App: Current location detected:', sdk.location);
  console.log('üîß Amplitude App: Available locations:', Object.keys(ComponentLocationSettings));

  const Component = useMemo(() => {
    console.log('üéØ Amplitude App: Determining component for location matching...');
    
    for (const [location, component] of Object.entries(ComponentLocationSettings)) {
      if (sdk.location.is(location)) {
        console.log(`‚úÖ Amplitude App: Location match found for ${location}, rendering component`);
        return component;
      }
    }
    
    console.warn('‚ö†Ô∏è Amplitude App: No matching location found, returning null');
    return null;
  }, [sdk.location]);

  console.log('üé® Amplitude App: Rendering component:', Component ? Component.name : 'null');
  
  return Component ? <Component /> : null;
};

export default App;
