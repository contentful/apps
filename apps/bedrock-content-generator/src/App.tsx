import { locations } from "@contentful/app-sdk";
import { useSDK } from "@contentful/react-apps-toolkit";
import ConfigScreen from "@locations/ConfigScreen";
import Dialog from "@locations/Dialog";
import Sidebar from "@locations/Sidebar";
import { useMemo } from "react";

const ComponentLocationSettings = () => ({
  [locations.LOCATION_APP_CONFIG]: ConfigScreen,
  [locations.LOCATION_DIALOG]: Dialog,
  [locations.LOCATION_ENTRY_SIDEBAR]: Sidebar,
});

const App = () => {
  const sdk = useSDK();

  const Component = useMemo(() => {
    const locations = ComponentLocationSettings();

    for (const [location, component] of Object.entries(locations)) {
      if (sdk.location.is(location)) {
        return component;
      }
    }
  }, [sdk.location]);

  return Component ? <Component /> : null;
};

export default App;
