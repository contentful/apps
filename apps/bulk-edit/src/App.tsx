import { useMemo, useEffect } from 'react';
import { locations, AppExtensionSDK } from '@contentful/app-sdk';
import Page from './locations/Page';
import { useSDK } from '@contentful/react-apps-toolkit';
import { i18n } from '@lingui/core';

const ComponentLocationSettings = {
  [locations.LOCATION_PAGE]: Page,
};

const App = () => {
  const sdk = useSDK<AppExtensionSDK>();

  // Initialize Lingui i18n for field editors that require it
  useEffect(() => {
    const defaultLocale = sdk.locales.default;
    if (!i18n.locale) {
      // Initialize with a default locale - field editors will use their own locales
      i18n.load(defaultLocale, {});
      i18n.activate(defaultLocale);
    }
  }, []);

  const Component = useMemo(() => {
    for (const [location, component] of Object.entries(ComponentLocationSettings)) {
      if (sdk.location.is(location)) {
        return component;
      }
    }
  }, [sdk.location]);

  return Component ? <Component /> : null;
};

export default App;
