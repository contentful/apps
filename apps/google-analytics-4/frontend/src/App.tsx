import * as Sentry from '@sentry/react';
import React, { useMemo } from 'react';
import { locations } from '@contentful/app-sdk';
import ConfigScreen from './locations/ConfigScreen';
import Field from './locations/Field';
import EntryEditor from './locations/EntryEditor';
import Dialog from './locations/Dialog';
import Sidebar from './locations/Sidebar';
import Page from './locations/Page';
import Home from './locations/Home';
import { useSDK } from '@contentful/react-apps-toolkit';

const ComponentLocationSettings = {
  [locations.LOCATION_APP_CONFIG]: ConfigScreen,
  [locations.LOCATION_ENTRY_FIELD]: Field,
  [locations.LOCATION_ENTRY_EDITOR]: EntryEditor,
  [locations.LOCATION_DIALOG]: Dialog,
  [locations.LOCATION_ENTRY_SIDEBAR]: Sidebar,
  [locations.LOCATION_PAGE]: Page,
  [locations.LOCATION_HOME]: Home,
};

const App = () => {
  const sdk = useSDK();

  const Component = useMemo(() => {
    for (const [location, component] of Object.entries(ComponentLocationSettings)) {
      if (sdk.location.is(location)) {
        // Set user information, as well as tags for contentful context
        Sentry.configureScope((scope) => {
          scope.clear();

          // user: "6iUaAfJg2ZoKe4oEYF7kxe"
          if (sdk.ids.user) scope.setUser({ id: sdk.ids.user });

          for (const [key, value] of Object.entries(sdk.ids)) {
            // app: "5eHTUt9pILTGjYk3VFE9ta"
            // contentType: "product"
            // entry: "4w6Hx5RgIO2uiNcgxpqj6l"
            // environment: "gary.hepting"
            // field: "sku"
            // organization: "6xdLsz6lCsk0yPOccSsDK7"
            // space: "30x8uoqewkkz"
            // user: "6iUaAfJg2ZoKe4oEYF7kxe"
            if (value) scope.setTag(`x-contentful-${key}`, value);
          }

          // location: "app-config"
          if (location) scope.setTag('x-contentful-location', location);
        });

        return component;
      }
    }
  }, [sdk.location, sdk.ids]);

  return Component ? <Component /> : null;
};

export default App;
