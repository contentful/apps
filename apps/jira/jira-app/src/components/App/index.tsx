import React from 'react';
import { AppExtensionSDK } from '@contentful/app-sdk';
import Config from './Config';
import JiraSoftware from '../JiraSoftware';
import Auth from '../Auth';

interface Props {
  sdk: AppExtensionSDK;
}

/** The Jira App location component */
const App = ({ sdk }: Props) => (
  <div className="app">
    <div className="background" />
    <div className="body">
      <div className="config">
        <Auth notifyError={sdk.notifier.error} mode="config" setReady={sdk.app.setReady}>
          {(token: string, client, reset) => <Config token={token} sdk={sdk} reauth={reset} />}
        </Auth>
      </div>
    </div>
    <div className="logo">
      <JiraSoftware />
    </div>
  </div>
);

export default App;
