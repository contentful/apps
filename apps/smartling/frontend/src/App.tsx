import React from 'react';
import { locations, AppExtensionSDK, SidebarExtensionSDK } from '@contentful/app-sdk';
import AppConfig from './AppConfig';
import Sidebar from './Sidebar';

interface Props {
  sdk: AppExtensionSDK | SidebarExtensionSDK;
}

export default class App extends React.Component<Props> {
  render() {
    const { sdk } = this.props;

    if (sdk.location.is(locations.LOCATION_APP_CONFIG)) {
      return <AppConfig sdk={sdk as AppExtensionSDK} />;
    } else if (sdk.location.is(locations.LOCATION_ENTRY_SIDEBAR)) {
      return (
        <Sidebar
          sdk={sdk as SidebarExtensionSDK}
          projectId={(sdk.parameters.installation as SmartlingParameters).projectId}
        />
      );
    }

    return null;
  }
}
