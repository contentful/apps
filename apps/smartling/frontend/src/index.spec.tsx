/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import { render, fireEvent, cleanup, configure, wait } from '@testing-library/react';
import { locations, AppExtensionSDK } from '@contentful/app-sdk';
import 'whatwg-fetch';
import fetchMock from 'fetch-mock';
import App from './App';
import contentTypeResponse from './mockData/contentTypeResponse.json';
import entryMockResponse from './mockData/entryMockResponse.json';

configure({
  testIdAttribute: 'data-test-id'
});

let mockSdk: any;
const originalStorage = window.localStorage;

describe('App', () => {
  beforeEach(() => {
    Object.defineProperty(window, 'localStorage', { writable: true, value: originalStorage });
    mockSdk = {
      location: {
        is() {
          return false;
        }
      },
      parameters: {
        installation: {
          projectId: 'project-id-123'
        }
      },
      window: {
        startAutoResizer() {}
      },
      locales: {
        names: {
          'en-US': 'English (United State)',
          'de-DE': 'German (Germany)'
        }
      },
      ids: {
        app: 'smartling-app-id',
        entry: 'entry-123',
        space: 'space-123'
      },
      app: {
        async setReady() {},
        async onConfigure() {},
        async isInstalled() {
          return false;
        },
        async getParameters() {
          return null;
        },
        async onConfigurationCompleted() {}
      },
      space: {
        async getEditorInterfaces() {
          return {
            items: []
          };
        },
        async getContentTypes() {
          return contentTypeResponse;
        }
      },
      notifier: {
        error() {}
      }
    };
  });

  afterEach(() => {
    cleanup();
  });

  it('should load the AppConfig page and allow for installation', async () => {
    mockSdk.app.onConfigure = jest.fn();
    mockSdk.location.is = (location: string) => location === locations.LOCATION_APP_CONFIG;
    const wrapper = render(<App sdk={mockSdk as any} />);

    const projectInput = await wrapper.findByTestId('cf-ui-text-input');

    fireEvent.change(projectInput, { target: { value: 'project-id-123' } });

    const inputs = await wrapper.findAllByTestId('cf-ui-controlled-input');

    const authorCT = inputs[0];

    fireEvent.click(authorCT, {});

    await wait();

    expect(await mockSdk.app.onConfigure.mock.calls[0][0]()).toMatchSnapshot();
    expect(wrapper).toMatchSnapshot();
  });

  it('should fail installation if no projectId is provided', async () => {
    mockSdk.app.onConfigure = jest.fn();
    mockSdk.location.is = (location: string) => location === locations.LOCATION_APP_CONFIG;

    render(<App sdk={mockSdk as any} />);
    await wait();

    expect(await mockSdk.app.onConfigure.mock.calls[0][0]()).toBe(false);
  });

  it('should render the sidebar app with signing and request button', () => {
    mockSdk.location.is = (location: string) => location === locations.LOCATION_ENTRY_SIDEBAR;
    const wrapper = render(<App sdk={mockSdk as AppExtensionSDK} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should try to refresh then open the oauth modal', async () => {
    mockSdk.location.is = (location: string) => location === locations.LOCATION_ENTRY_SIDEBAR;
    window.open = jest.fn();

    fetchMock.get('/entry?spaceId=space-123&projectId=project-id-123&entryId=entry-123', {});
    fetchMock.get('/refresh?refresh_token=', 401);

    const wrapper = render(<App sdk={mockSdk as AppExtensionSDK} />);
    const oauthButton = await wrapper.findByTestId('open-dialog');

    oauthButton.click();
    await wait();

    expect(window.open).toHaveBeenCalledWith(
      '/openauth',
      '',
      'height=600,width=600,top=50,left=50'
    );
  });

  it('should get a new refresh token then get the entry', async () => {
    mockSdk.location.is = (location: string) => location === locations.LOCATION_ENTRY_SIDEBAR;
    window.open = jest.fn();

    Object.defineProperty(window, 'localStorage', {
      writable: true,
      value: {
        getItem: jest.fn((item: string) => {
          switch (item) {
            case 'token':
              return 'access-123';
            case 'refreshToken':
              return 'refresh-123';
            default:
              return null;
          }
        }),
        setItem: jest.fn(),
        removeItem: jest.fn()
      }
    });

    fetchMock.get(
      '/refresh?refresh_token=',
      { access_token: 'access-123' },
      { overwriteRoutes: true }
    );
    fetchMock.get(
      '/entry?spaceId=space-123&projectId=project-id-123&entryId=entry-123',
      entryMockResponse,
      { overwriteRoutes: true }
    );

    const wrapper = render(<App sdk={mockSdk as AppExtensionSDK} />);
    await wait();

    expect(wrapper).toMatchSnapshot();
  });

  it('should show an error message when getting a general error from Smartling', async () => {
    mockSdk.location.is = (location: string) => location === locations.LOCATION_ENTRY_SIDEBAR;

    Object.defineProperty(window, 'localStorage', {
      writable: true,
      value: {
        getItem: jest.fn((item: string) => {
          switch (item) {
            case 'token':
              return 'access-123';
            case 'refreshToken':
              return 'refresh-123';
            default:
              return null;
          }
        }),
        setItem: jest.fn(),
        removeItem: jest.fn()
      }
    });

    fetchMock.get(
      '/refresh?refresh_token=',
      401,
      { overwriteRoutes: true }
    );
    fetchMock.get(
      '/entry?spaceId=space-123&projectId=project-id-123&entryId=entry-123',
      {},
      { overwriteRoutes: true }
    );

    const wrapper = render(<App sdk={mockSdk as AppExtensionSDK} />);
    await wait();

    expect(wrapper).toMatchSnapshot();
  });
});
