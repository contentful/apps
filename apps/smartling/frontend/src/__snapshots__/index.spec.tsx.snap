// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should load the AppConfig page and allow for installation 1`] = `
{
  "parameters": {
    "projectId": "project-id-123",
  },
  "targetState": {
    "EditorInterface": {},
  },
}
`;

exports[`App > should render the sidebar app with signing and request button 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="css-dt2tyy signin"
        data-test-id="open-dialog"
        type="button"
      >
        <span
          class="css-40g50j"
        >
          Sign in with Smartling
        </span>
      </button>
      <br />
      <button
        class="css-1ka7c9s request-translation"
        data-test-id="request-translation"
        type="button"
      >
        <span
          class="css-40g50j"
        >
          <svg
            aria-hidden="true"
            class="css-1jrff5x"
            data-test-id="cf-ui-icon"
            role="img"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
           Request Translation
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="css-dt2tyy signin"
      data-test-id="open-dialog"
      type="button"
    >
      <span
        class="css-40g50j"
      >
        Sign in with Smartling
      </span>
    </button>
    <br />
    <button
      class="css-1ka7c9s request-translation"
      data-test-id="request-translation"
      type="button"
    >
      <span
        class="css-40g50j"
      >
        <svg
          aria-hidden="true"
          class="css-1jrff5x"
          data-test-id="cf-ui-icon"
          role="img"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
         Request Translation
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App > should show an error message when getting a general error from Smartling 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="smartling-status"
      >
        <div>
          <p
            class="info-p css-ki4iwd"
            data-test-id="cf-ui-paragraph"
          >
            Smartling status
          </p>
        </div>
        <div>
          <div
            class="job-status css-1jskwfg"
            data-test-id="cf-ui-badge"
          >
            <span
              class="css-1f8vwjn"
              data-test-id="cf-ui-caption"
            >
              Disconnected
            </span>
          </div>
        </div>
      </div>
      <button
        class="css-1ka7c9s request-translation"
        data-test-id="request-translation"
        type="button"
      >
        <span
          class="css-40g50j"
        >
          <svg
            aria-hidden="true"
            class="css-1jrff5x"
            data-test-id="cf-ui-icon"
            role="img"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
           Request Translation
        </span>
      </button>
      <article
        class="general-error css-ep96ao"
        data-test-id="cf-ui-note"
      >
        <svg
          aria-hidden="true"
          class="css-77587x"
          data-test-id="cf-ui-icon"
          role="img"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
          />
        </svg>
        <div
          class="css-j7qwjs"
        >
          <h2
            class="css-1fgrah0"
            data-test-id="cf-ui-heading"
          >
            Connection error
          </h2>
          <div
            class="css-n69953"
          >
            Please ensure that you have access to the connected Smartling project.
            <br />
            <a
              class="css-1wm2dh8"
              data-test-id="cf-ui-text-link"
              href="https://contentful.com/developers/docs/extensibility/apps/smartling/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                class="css-x1sij0"
              >
                <span>
                  View documentation
                </span>
              </span>
            </a>
          </div>
        </div>
      </article>
      <p
        class="smartling-more-info css-ki4iwd"
        data-test-id="cf-ui-paragraph"
      >
        <a
          class="css-1wm2dh8"
          data-test-id="cf-ui-text-link"
          href="https://help.smartling.com/hc/en-us/articles/360000546974-Contentful-Connector-Overview"
          rel="noopener"
          target="_blank"
        >
          <span
            class="css-x1sij0"
          >
            <span>
              Learn how Smartling works with ContentfulÂ 
              <svg
                aria-hidden="true"
                class="css-1q4rrbp"
                data-test-id="cf-ui-icon"
                role="img"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 19H5V5h7V3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </span>
          </span>
        </a>
      </p>
    </div>
  </body>,
  "container": <div>
    <div
      class="smartling-status"
    >
      <div>
        <p
          class="info-p css-ki4iwd"
          data-test-id="cf-ui-paragraph"
        >
          Smartling status
        </p>
      </div>
      <div>
        <div
          class="job-status css-1jskwfg"
          data-test-id="cf-ui-badge"
        >
          <span
            class="css-1f8vwjn"
            data-test-id="cf-ui-caption"
          >
            Disconnected
          </span>
        </div>
      </div>
    </div>
    <button
      class="css-1ka7c9s request-translation"
      data-test-id="request-translation"
      type="button"
    >
      <span
        class="css-40g50j"
      >
        <svg
          aria-hidden="true"
          class="css-1jrff5x"
          data-test-id="cf-ui-icon"
          role="img"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
         Request Translation
      </span>
    </button>
    <article
      class="general-error css-ep96ao"
      data-test-id="cf-ui-note"
    >
      <svg
        aria-hidden="true"
        class="css-77587x"
        data-test-id="cf-ui-icon"
        role="img"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
        />
      </svg>
      <div
        class="css-j7qwjs"
      >
        <h2
          class="css-1fgrah0"
          data-test-id="cf-ui-heading"
        >
          Connection error
        </h2>
        <div
          class="css-n69953"
        >
          Please ensure that you have access to the connected Smartling project.
          <br />
          <a
            class="css-1wm2dh8"
            data-test-id="cf-ui-text-link"
            href="https://contentful.com/developers/docs/extensibility/apps/smartling/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              class="css-x1sij0"
            >
              <span>
                View documentation
              </span>
            </span>
          </a>
        </div>
      </div>
    </article>
    <p
      class="smartling-more-info css-ki4iwd"
      data-test-id="cf-ui-paragraph"
    >
      <a
        class="css-1wm2dh8"
        data-test-id="cf-ui-text-link"
        href="https://help.smartling.com/hc/en-us/articles/360000546974-Contentful-Connector-Overview"
        rel="noopener"
        target="_blank"
      >
        <span
          class="css-x1sij0"
        >
          <span>
            Learn how Smartling works with ContentfulÂ 
            <svg
              aria-hidden="true"
              class="css-1q4rrbp"
              data-test-id="cf-ui-icon"
              role="img"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 19H5V5h7V3H5a2 2 0 00-2 2v14a2 2 0 002 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </span>
      </a>
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
